homeassistant/components/nordpool/__init__.py

```diff
@@ -46,3 +52,21 @@ async def async_setup_entry(hass: HomeAssistant, entry: NordPoolConfigEntry) ->
 async def async_unload_entry(hass: HomeAssistant, entry: NordPoolConfigEntry) -> bool:
     """Unload Nord Pool config entry."""
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def cleanup_device(hass: HomeAssistant, entry: NordPoolConfigEntry) -> None:
+    """Cleanup device and entities."""
+    device_reg = dr.async_get(hass)
+    entity_reg = er.async_get(hass)
+
+    entries = dr.async_entries_for_config_entry(device_reg, entry.entry_id)
+    for area in entry.data[CONF_AREAS]:
+        for _entry in entries:
+            if _entry.identifiers == {(DOMAIN, area)}:
+                continue
+
+            LOGGER.debug("Removing device %s", _entry.name)
+            entities = er.async_entries_for_device(entity_reg, _entry.id)
+            for _entity in entities:
+                entity_reg.async_remove(_entity.entity_id)
+            device_reg.async_clear_config_entry(entry.entry_id)
```

Comment:
Normally removing the config entry from the device is preferred (afaik)
```python
device_registry.async_update_device(
                        device_id=device.id,
                        remove_config_entry_id=self.config_entry.entry_id,
                    )
```
is there a reason you chose to not do this here?

Reply:
I think that's what I am doing (but with a different method)?